RELEASE.md

```diff
@@ -0,0 +1,304 @@
+# Release process
+
+The release process starts with one or more tagged release candidates,
+when no more blocking issues needs to be fixed a final tag is created
+and the final release is rolled.
+
+confluent-kafka-dotnet uses semver versioning and loosely follows
+librdkafka's version, e.g. v0.11.x for the final release and
+v0.11.x-RC3 for the 3rd v0.11.x release candidate.
```

Comment:
it's confusing to specify 'x' in the final release tag here, since an actual one wouldn't and 0.11.x is the name of a branch ("do we release a branch?!?! huh?"). maybe:

"confluent-kafka-dotnet uses semver versioning and loosely follows the referenced librdkafka.redist version. For example, a final release tag might be v0.11.0 and the third release candidate for the same release would be tagged v0.11.0-RC3."